### PR TITLE
Added additional logging for SQL Engine Pushdown execution.

### PR DESCRIPTION
**Pull request by fernst**
Improved error handling by catching all throwables

### CODE REVIEW

File: cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/BatchSQLEngineAdapter.java [hunk 1]
```diff
@@ -188,11 +191,14 @@ public SQLDataset pushInternal(String datasetName,
...
       try {
+        LOG.debug("Starting pull for dataset '{}'", job.getDatasetName());
         waitForJobAndHandleExceptionInternal(job);
         JavaRDD<T> result = pullInternal(job.waitFor(), jsc);
+        LOG.debug("Completed pull for dataset '{}'", job.getDatasetName());
         future.complete(result);
-      } catch (Exception e) {
-        future.completeExceptionally(e);
+      } catch (Throwable t) {
+        LOG.warn("Error when executing pull for dataset '{}'", job.getDatasetName());
```
Comment (tivv) [inline cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/BatchSQLEngineAdapter.java]: This does not tells which error. Also it duplicate all the logging in it `wait`. Could you add throwable and make one of it DEBUG?

Comment (fernst) [reply cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/BatchSQLEngineAdapter.java]: Ah, good catch. I'll remove these logs

File: cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/BatchSQLEngineAdapter.java [hunk 2]
```diff
@@ -293,8 +298,9 @@ public boolean canJoin(String datasetName,
         }
 
         joinInternal(future, joinRequest);
-      } catch (Exception e) {
-        future.completeExceptionally(e);
+        LOG.debug("Completed join for dataset '{}'", datasetName);
+      } catch (Exception t) {
```
Comment (tivv) [inline cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/BatchSQLEngineAdapter.java]: You need throwable here as well, don't you?

Comment (fernst) [reply cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/BatchSQLEngineAdapter.java]: My bad, you are correct

**Review by tivv — APPROVED** [review]

**Pull request merged**